iocore/net/SSLUtils.cc

```diff
@@ -418,6 +418,61 @@ PerformAction(Continuation *cont, const char *servername)
   return SSL_TLSEXT_ERR_OK;
 }
 
+#if TS_USE_HELLO_CB
+// Pausable callback
+static int
+ssl_client_hello_callback(SSL *s, int *al, void *arg)
+{
+  const char *servername = nullptr;
+  const unsigned char *p;
+  size_t remaining, len;
+  if (SSL_client_hello_get0_ext(s, TLSEXT_TYPE_server_name, &p, &remaining) || remaining <= 2) {
+    // Parse to get to the name, originally from test/handshake_helper.c in openssl tree
+    /* Extract the length of the supplied list of names. */
+    len = (*(p++) << 8);
+    len += *(p++);
+    if (len + 2 == remaining) {
+      remaining = len;
+      /*
+       * The list in practice only has a single element, so we only consider
+       * #endif
+       * the first one.
+       */
+      if (remaining != 0 && *p++ == TLSEXT_NAMETYPE_host_name) {
+        remaining--;
+        /* Now we can finally pull out the byte array with the actual hostname. */
+        if (remaining > 2) {
+          len = (*(p++) << 8);
```

Comment:
Why not `*(p++) << 8`?

Reply:
And similarly too up on line 432, but seems a little nick picky :).

Reply:
Because I was stealing code from the openssl project.  Will tidy up.